Creational/Builder/Director.php

```diff
@@ -20,9 +20,12 @@ class Director
     public function build(BuilderInterface $builder)
     {
         $builder->createVehicle();
-        $builder->addDoors();
         $builder->addEngine();
         $builder->addWheel();
+        
+        if ($builder instanceof CarBuilder) {
```

Comment:
Maybe this is a bad example, as all implementations of BuilderInterface should provide only the methods of the interface to be used here. Checking for instanceof CarBuilder here seems to rather dirty. What do you think if we remove BikeBuilder and replace it with a TruckBuilder?